pyo3_special_method_derive_macro/src/str_repr.rs

```diff
@@ -107,29 +127,73 @@ fn generate_fmt_impl_for_struct(
         .iter()
         .enumerate()
         .map(|(i, field)| {
+            let display_attr = {
+                let mut display_attr = None;
+
+                for attr in &field.attrs {
+                    let path = attr.path();
+                    if path.is_ident(ATTR_NAMESPACE_FORMATTER) {
+                        display_attr = Some(attr);
+                    }
+                }
+
+                display_attr
+            };
+
+            let mut variant_fmt = quote! { #DEFAULT_ELEMENT_FORMATTER };
+            if let Some(display_attr) = display_attr {
+                if let Some(formatter) = find_display_attribute(display_attr) {
+                    variant_fmt = formatter;
+                }
+            }
+
+            let formatters = variant_fmt.to_string().matches("{}").count()
+                - variant_fmt.to_string().matches("{{}}").count();
+            if formatters > 1 {
+                panic!("Specify 1 (variant), or 0 formatters in the format string.");
+            };
```

Comment:
I think we syn we can do a compile error but otherwise alright

Reply:
Updated to use that!